internal/pkg/levee/testdata/src/example.com/tests/propagation/tests.go

```diff
@@ -20,21 +20,77 @@ import (
 	"example.com/core"
 )
 
-func Identity(arg interface{}) interface{} {
-	return arg
+func main() {
+	TestSinkWrapperWrapper(core.Source{})
+	TestSinkWrapper(core.Source{})
+	TestSinkWrapperTwoArgs(core.Source{})
+	TestOneArgSinkWrapper(core.Source{})
+	TestSinkWrapperSlice(core.Source{})
+	TestSinkWrapperSpread(core.Source{})
+	TestStringify(core.Source{})
 }
 
-func TestIdentityPropagator(s core.Source) {
-	i := Identity(s)
-	core.Sink(i)           // want "a source has reached a sink"
-	core.Sink(Identity(s)) // want "a source has reached a sink"
+func TestSinkWrapperWrapper(s core.Source) {
```

Comment:
How would this play with sanitizers?
Could we add a test? Even if it is a failing test.

Reply:
It would fail to recognize that the pointed-to value has been sanitized and produce a report. :/

Reply:
Do you think that asking the pointer analysis to also produce a CallGraph somehow could help us with this?

Reply:
Hmm. That is a good question.

I don't think so. Given an `ssa.Value`, we can get the function it was defined in by using its `Parent()` function. By traversing the call graph we might be able to determine e.g. that the function that contains the source value calls a sanitizer at some point, but we would still need to look at the function's body to determine whether sanitization is done properly. So I don't think `pointer` analysis would help us much, we would need to do a lot of work outside of the analysis.